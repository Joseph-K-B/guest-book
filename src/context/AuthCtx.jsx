import React, { createContext, useState } from "react"

const authCtx = createContext();

const AuthProvider = ({ children }) => {
  const [user, setUser] = useState(null);

  // This would probably be the sign up function 
  // b/c it includes both username and email
  // 
  const login = (username, email, password) => {
    const authConfirmed = (
      username === process.env.REACT_APP_AUTH_USER &&
      email === process.env.REACT_APP_AUTH_EMAIL &&
      password === process.env.REACT_APP_AUTH_PASSWORD
      )
      if(authConfirmed) 
        setUser({ username, email });
      return authConfirmed;
  };

  const logout = (cb) => {
    setUser(null);
    // cb();
  }

  return (
    <authCtx.Provider value={{user, setUser, login, logout}}>
      {children}
    </authCtx.Provider>
  );
};

export {authCtx, AuthProvider};