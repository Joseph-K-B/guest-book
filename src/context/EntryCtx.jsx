import { createContext, useContext, useState } from "react";

const NoteCtx = createContext();

const NoteProvider = ({children}) => {
    const [note, setNote] = useState('');

    return <NoteCtx.Provider value={{ note, setNote }}>{children}</NoteCtx.Provider>
}

const useNote = () => {
    const ctx = useContext(NoteCtx);

    if(ctx === undefined) {
        throw new Error('useNote hook can only be called within child components of NoteCtx Provider');
    }
    return ctx;
}

export { NoteProvider, useNote}