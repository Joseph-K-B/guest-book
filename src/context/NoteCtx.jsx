import React from 'react';
import { createContext, useContext, useState } from "react";

const NoteCtx = createContext();

const NoteProvider = ({ children }) => {
    const [notes, setNotes] = useState([]);

    return (
        <NoteCtx.Provider value={{ notes, setNotes }}>{children}</NoteCtx.Provider>
    )
}

const useNote = () => {
    const ctx = useContext(NoteCtx);

    if(ctx === undefined) {
        throw new Error('useNote hook can only be called within child components of NoteCtx Provider');
    }
    return ctx;
}

export { NoteProvider, useNote}