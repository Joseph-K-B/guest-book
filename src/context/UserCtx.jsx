import React from 'react';
import { createContext, useContext, useState } from "react"

const UserCtx = createContext();

const UserProvider = ({children}) => {
    const [name, setName] = useState('');

    return <UserCtx.Provider value={{ name, setName }}>{children}</UserCtx.Provider>
}

const useUser = () => {
    const ctx = useContext(UserCtx);

    if(ctx === undefined) {
        throw new Error('useUser custom hook must be called within child components of UserCtx Provider');
    }
    return ctx;
}

export { UserProvider, useUser }