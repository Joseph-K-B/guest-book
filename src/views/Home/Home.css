.mainSect {
  display: flex;
  flex-direction: column-reverse;
  justify-items: center;
  align-items: center;
}

.homeImg {
  max-height: 20rem;
  max-width: 20rem;
}

.introDiv {
  display: flex;
  flex-direction: column;
  margin: 7%;
  align-items: center;
  text-align: center;
  max-height: 30rem;
  max-width: 15rem;
  color: white;
  border: white solid 0.5rem;
}


@media (min-width: 700px) {
  .mainSect {
    display: grid;
    grid-template-columns: 1fr 1.75fr;  
  }
  .homeImg {
    max-height: 70vh;
    margin: 10%;
  }
}


