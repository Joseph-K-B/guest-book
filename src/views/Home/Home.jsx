import { useState } from "react"
import { EntryForm } from "../../components/entry/EntryForm";
import { NoteProvider } from "../../context/NoteCtx";


const Home = () => {
    // const [loading, setLoading] = useState(true);

    // if(loading) return <h1>Loading...</h1>
    return (
        <NoteProvider>
            <EntryForm />
        </NoteProvider>
    )
}

export default Home