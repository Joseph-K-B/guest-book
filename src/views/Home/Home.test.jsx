import React from 'react';
import { screen, render } from '@testing-library/react';
import { NoteProvider } from '../../context/NoteCtx';
import { UserProvider } from '../../context/UserCtx';
import Home from './Home';

it('renders Home component and allows user to input name & message', () => {
    const { container } = render(
        <UserProvider>
            <NoteProvider>
                <Home/>
            </NoteProvider>
        </UserProvider>
    );

    const nameInput = screen.getByLabelText('name-input');
    const msgInput = screen.getByLabelText('note-input');
    const submitBtn = screen.getByRole('button');

    expect(nameInput).toBeInTheDocument();
    expect(msgInput).toBeInTheDocument();
    expect(submitBtn).toBeInTheDocument();
    expect(container).toMatchSnapshot();
})