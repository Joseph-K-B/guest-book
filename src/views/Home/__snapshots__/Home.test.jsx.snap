// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home component and allows user to input name & message 1`] = `
<div>
  <h3>
    Sign here and leave a note
  </h3>
  <form>
    <div>
      <input
        aria-label="name-input"
        placeholder="name"
        type="text"
        value=""
      />
    </div>
    <input
      aria-label="note-input"
      placeholder="note"
      type="text"
      value=""
    />
    <button>
      Submit
    </button>
    
  </form>
  <ul />
</div>
`;
